ec2/ec2.go

```diff
@@ -138,20 +139,21 @@ func (ec2 *EC2) query(params map[string]string, resp interface{}) error {
 	}
 	sign(ec2.Auth, "GET", endpoint.Path, params, endpoint.Host)
 	endpoint.RawQuery = multimap(params).Encode()
-	if debug {
-		log.Printf("get { %v } -> {\n", endpoint.String())
-	}
+	log.Debugf("GET %v", endpoint.String())
 	r, err := ec2.httpClient.Get(endpoint.String())
 	if err != nil {
 		return err
 	}
 	defer r.Body.Close()
 
-	if debug {
-		dump, _ := httputil.DumpResponse(r, true)
-		log.Printf("response:\n")
-		log.Printf("%v\n}\n", string(dump))
+	dump := []byte{}
+	if tracingEnabled() {
+		dump, _ = httputil.DumpResponse(r, true)
+	} else {
+		dump, _ = httputil.DumpResponse(r, false)
```

Comment:
Good point. Ideally we would be able to wrap this whole thing into if log.isDebugEnabled(). Haven't found a way to do that with logrus yet.

Just printing the response headers is not that expensive. On the other hand printing body can be quite expensive depending on the body. But we will only be printing the body when we know we are doing debugging using TRACE